packages/api/test/e2e/promise-metadata.spec.js

```diff
@@ -0,0 +1,68 @@
+    const genesisHash = '0x8579ded4438b26597e188cda8579fdd6abf09b74ab672e604b9584cd8b4c3d0b';
+    const specVersion = '62';
+    const prebundles = {};
+    const key = `${genesisHash}-${specVersion}`;
+    prebundles[key] = rpcData;
+    const api = await ApiPromise.create({
+      prebundles
+    });
+
+    expect(api.genesisHash).toBeDefined();
+    expect(api.runtimeMetadata).toBeDefined();
+    expect(api.runtimeVersion).toBeDefined();
+    expect(api.rpc).toBeDefined();
+    expect(api.query).toBeDefined();
+    expect(api.tx).toBeDefined();
+    expect(api.derive).toBeDefined();
+  });
+
+  it('Create API instance without metadata and makes the runtime, rpc, state & extrinsics available', async() => {
+    const prebundles = {};
+    const api = await ApiPromise.create({
+      prebundles
+    });
+
+    expect(api.genesisHash).toBeDefined();
+    expect(api.runtimeMetadata).toBeDefined();
+    expect(api.runtimeVersion).toBeDefined();
+    expect(api.rpc).toBeDefined();
+    expect(api.query).toBeDefined();
+    expect(api.tx).toBeDefined();
+    expect(api.derive).toBeDefined();
+  });
+
+  it('Create API instance with incorect metadata map and makes the runtime, rpc, state & extrinsics available', async() => {
+    const rpcData = 'invalid data';
+    const genesisHash = '0x8579ded4438b26597e188cda8579fdd6abf09b74ab672e604b9584cd8b4c3d0b';
```

Comment:
It's kind of annoying that each time I want to run this test with my local node, I need to change this line manually. Maybe we could create a promise-alex.spec.js or promise-ee.spec.js.

Edit: on top of an e2e test, we could actually add a unit test, with the mockProvider (inside rpc-provider) that will return these pre-defined values (hash, rpcData, specVersion)

Reply:
Thanks..Changed as per suggestion.